vxquery-core/src/test/java/org/apache/vxquery/datamodel/ObjectByteTest.java

```diff
@@ -0,0 +1,261 @@
+                    + ValueTag.XS_STRING_TAG + " Got valuetag: " + tvp.getTag());
+        }
+
+        if (!op.getValue(tvpKey2, tvp)) {
+            Assert.fail("Value not found for the given key: price");
+        }
+        if (!FunctionHelper.arraysEqual(tvp, tvpValue2)) {
+            Assert.fail("Value is incorrect for the given key. Expected: 12.5 with valuetag: " + ValueTag.XS_DOUBLE_TAG
+                    + " Got valuetag: " + tvp.getTag());
+        }
+
+        if (!op.getValue(tvpKey3, tvp)) {
+            Assert.fail("Value not found for the given key: properties");
+        }
+        if (!FunctionHelper.arraysEqual(tvp, tvpValue3)) {
+            Assert.fail("Value is incorrect for the given key. Expected: 100 with valuetag: " + ValueTag.XS_LONG_TAG
+                    + " Got valuetag: " + tvp.getTag());
+        }
+    }
+
+    @Test
+    public void testItemNotExistObject() {
+        // Build empty test Object
+        abvsResult.reset();
+        try {
+            ob.reset(abvsResult);
+            ob.finish();
+        } catch (IOException e) {
+            Assert.fail("Test failed to write the Object pointable.");
+        }
+        tvp.set(abvsResult);
+        tvp.getValue(op);
+        try {
+            getTaggedValuePointable("key", false, tvpKey1);
+            if (op.getValue(tvpKey1, tvp)) {
+                Assert.fail("key not in object. Expected: false Got: true");
+            }
+        } catch (IOException e) {
+            Assert.fail("Test failed to write the object pointable.");
+        }
+
+        // Build test object
+        try {
+            // Add two items
+            abvsResult.reset();
+            ob.reset(abvsResult);
+            getTaggedValuePointable("name", false, tvpKey1);
+            getTaggedValuePointable("A green door", tvpValue1);
+            ob.addItem(tvpKey1, tvpValue1);
+            getTaggedValuePointable("price", false, tvpKey2);
+            getTaggedValuePointable(12.5, tvpValue2);
+            ob.addItem(tvpKey2, tvpValue2);
+            ob.finish();
+        } catch (IOException e) {
+            Assert.fail("Test failed to write the object pointable.");
+        }
+
+        tvp.set(abvsResult);
+        tvp.getValue(op);
+        try {
+            getTaggedValuePointable("key", false, tvpKey3);
+            if (op.getValue(tvpKey3, tvp)) {
+                Assert.fail("key not in object. Expected: false Got: true");
+            }
+
+            if (!op.getValue(tvpKey1, tvp)) {
+                Assert.fail("key in object. Expected: true Got: false");
+            }
+        } catch (IOException e) {
+            Assert.fail("Test failed to write the object pointable.");
+        }
+    }
+
+    public boolean compareKeys(IPointable tvp1, IPointable tvp2) {
+        return FunctionHelper.arraysEqual(tvp1.getByteArray(), tvp1.getStartOffset() + 1, tvp1.getLength() - 1,
```

Comment:
Something like this...

private IPointable vp = VoidPointable.FACTORY.createPointable();
public boolean compareKeys(TaggedValuePointable tvp1, IPointable tvp2) {
    tvp1.getValue(vp)
    return FunctionHelper.arraysEqual(vp.getByteArray(),
vp.getStartOffset(), vp.getLength(),
        tvp2.getByteArray(), tvp2.getStartOffset(), tvp2.getLength());
}

On Wed, May 18, 2016 at 8:27 PM, Riyafa Abdul Hameed
<notifications@github.com> wrote:
> In
> vxquery-core/src/test/java/org/apache/vxquery/datamodel/ObjectByteTest.java:
>
>> +        try {
>> +            getTaggedValuePointable("key", false, tvpKey3);
>> +            if (op.getValue(tvpKey3, tvp)) {
>> +                Assert.fail("key not in object. Expected: false Got:
>> true");
>> +            }
>> +
>> +            if (!op.getValue(tvpKey1, tvp)) {
>> +                Assert.fail("key in object. Expected: true Got: false");
>> +            }
>> +        } catch (IOException e) {
>> +            Assert.fail("Test failed to write the object pointable.");
>> +        }
>> +    }
>> +
>> +    public boolean compareKeys(IPointable tvp1, IPointable tvp2) {
>> +        return FunctionHelper.arraysEqual(tvp1.getByteArray(),
>> tvp1.getStartOffset() + 1, tvp1.getLength() - 1,
>
> I am using this method because the pointable returned from getKeys has tag
> and the key I have created don't have tag. This is to compare both. Why use
> getValue method of TaggedValuePointable in this case? Sorry about the
> confusion
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub